addons/survey/models/survey_question.py

```diff
@@ -710,35 +724,60 @@ class SurveyQuestionAnswer(models.Model):
     is_correct = fields.Boolean('Correct')
     answer_score = fields.Float('Score', help="A positive score indicates a correct choice; a negative or null score indicates a wrong answer")
 
+    _sql_constraints = [
+        ('value_not_empty', "CHECK (value IS NOT NULL OR value_image_filename IS NOT NULL)",
+         'Suggested answer value must not be empty (a text and/or an image must be provided).'),
+    ]
+
     @api.depends('value', 'question_id.question_type', 'question_id.title', 'matrix_question_id')
     def _compute_display_name(self):
-        """Render an answer name as "Question title : Answer value", making sure it is not too long.
+        """Render an answer name as "Question title : Answer label value", making sure it is not too long.
 
         Unless the answer is part of a matrix-type question, this implementation makes sure we have
         at least 30 characters for the question title, then we elide it, leaving the rest of the
         space for the answer.
         """
         for answer in self:
             if not answer.question_id or answer.question_id.question_type == 'matrix':
-                answer.display_name = answer.value
+                answer.display_name = answer.value_label
                 continue
             title = answer.question_id.title or _("[Question Title]")
-            n_extra_characters = len(title) + len(answer.value) + 3 - self.MAX_ANSWER_NAME_LENGTH  # 3 for `" : "`
+            answer_label = answer.value_label or ''
+            n_extra_characters = len(title) + len(answer_label) + 3 - self.MAX_ANSWER_NAME_LENGTH  # 3 for `" : "`
             if n_extra_characters <= 0:
-                answer.display_name = f'{title} : {answer.value}'
+                answer.display_name = f'{title} : {answer_label}'
             else:
                 answer.display_name = shorten(
-                    f'{shorten(title, max(30, len(title) - n_extra_characters), placeholder="...")} : {answer.value}',
+                    f'{shorten(title, max(30, len(title) - n_extra_characters), placeholder="...")} : '
+                    f'{answer_label}',
                     self.MAX_ANSWER_NAME_LENGTH,
                     placeholder="..."
                 )
 
+    @api.depends('question_id.suggested_answer_ids', 'value')
+    def _compute_value_label(self):
+        """ Compute the label as the value if not empty or a letter representing the index of the answer otherwise. """
+        current_question_id, answer_idx = False, 0
+        assert self._order.startswith('question_id')
+        answers_with_question = self.env['survey.question.answer'].search([('question_id', 'in', self.question_id.ids)])
+        for answer in answers_with_question:
+            if answer.question_id != current_question_id:
+                current_question_id, answer_idx = answer.question_id, 0
+            if answer in self:
+                answer.value_label = answer.value or (chr(65 + answer_idx) if answer_idx < 27 else '')
```

Comment:
I would try to be a bit less "optimized" and use recordsets, like
```suggestion
        for answer in self:
            value_label = answer.value
            # using image -> use a letter to represent the value
            if not value_label:
                answer_idx = answer.question_id.suggested_answer_ids.ids.index(answer.id)
                value_label = chr(65 + answer_idx) if answer_idx < 27 else ''
            answer.value_label = value_label
```

Reply:
@tde-banana-odoo did you see my proposal too?